lasagne/layers/conv.py

```diff
@@ -249,7 +249,7 @@ def get_output_for(self, input, input_shape=None, **kwargs):
                                       filter_shape=self.get_W_shape(),
                                       border_mode='full')
             shift = (self.filter_size[0] - 1) // 2
-            conved = conved[:, :, shift:input.shape[2] + shift]
+            conved = conved[:, :, shift:-shift]
```

Comment:
Ah yes, this discussion is coming back to me now... I was basing my opinion on a vague memory of [this comment](https://github.com/Lasagne/Lasagne/blob/102203f8ef23ae573bb20783cbe7cf208e9d1f8c/lasagne/layers/dnn.py#L262), but it seems like at some point that changed. 

I can start a new PR to update the tests and see what else needs to be fixed, but probably won't have time today.